### ISSUE TITLE
Potential refactoring of ExportManager.GetPictures(Product) method

### ISSUE THREAD

**Issue by davidandrewpowell** [nopSolutions/nopCommerce#3885]
nopCommerce version: 4.20 beta

Please correct me if I am wrong but I believe that the following `ExportManager.GetPictures(Product)` method:

```
protected virtual string[] GetPictures(Product product)
{
    //pictures (up to 3 pictures)
    string picture1 = null;
    string picture2 = null;
    string picture3 = null;
    var pictures = _pictureService.GetPicturesByProductId(product.Id, 3);
    for (var i = 0; i < pictures.Count; i++)
    {
        var pictureLocalPath = _pictureService.GetThumbLocalPath(pictures[i]);
        switch (i)
        {
            case 0:
                picture1 = pictureLocalPath;
                break;
            case 1:
                picture2 = pictureLocalPath;
                break;
            case 2:
                picture3 = pictureLocalPath;
                break;
        }
    }

    return new[] { picture1, picture2, picture3 };
}

```
can be simplified as:

```
protected virtual string[] GetPictures(Product product)
{
    var pictures = _pictureService.GetPicturesByProductId(product.Id, 3);
    return pictures.Select(picture => _pictureService.GetThumbLocalPath(picture)).ToArray();
}
```

Going further, it seems this method is only called when creating a `PropertyByName` in the `ExportManager.ExportProductsToXlsx(IEnumerable<Product>)` method:

```
...
new PropertyByName<Product>("Picture1", p => GetPictures(p)[0]),
new PropertyByName<Product>("Picture2", p => GetPictures(p)[1]),
new PropertyByName<Product>("Picture3", p => GetPictures(p)[2])
...
```

This means that 3 pictures are returned from the method each time it is called when only one is needed.  Instead, we can just return a single picture to give:

```
protected virtual string GetPicture(Product product, int pictureIndex)
{
    var pictures = _pictureService.GetPicturesByProductId(product.Id, 3);
    return _pictureService.GetThumbLocalPath(pictures[pictureIndex]);
}

...
new PropertyByName<Product>("Picture1", p => GetPicture(p, 0),
new PropertyByName<Product>("Picture2", p => GetPicture(p, 1),
new PropertyByName<Product>("Picture3", p => GetPicture(p, 2)
...
```

Note the renaming of the method above from `GetPictures` to `GetPicture`.  Note also that we have reduced the number of times that we call `_pictureService.GetThumbLocalPath(Picture)`.

And finally, if we look at what the method is returning we will see that it does not return a picture (as implied by the name) but instead returns a picture thumbnail path or a null value if no picture is found.  Therefore, we should rename the method appropriately to reflect this.  I suggest the following:

```
protected virtual string GetPictureThumbLocalPathOrNull(Product product, int pictureIndex)
{
    var pictures = _pictureService.GetPicturesByProductId(product.Id, 3);
    return _pictureService.GetThumbLocalPath(pictures[pictureIndex]);
}
```

Note that the comments above the method will also need to be updated.

**Comment by roman-gherman** [comment]
Refactored in this pull request for nopCommerce version 4.30
https://github.com/nopSolutions/nopCommerce/pull/4815

**Comment by skoshelev** [comment]
Closed #3885
